chain/chain/src/flat_storage_resharder.rs

```diff
@@ -768,11 +862,14 @@ fn shard_split_handle_key_value(
         }
         col::DELAYED_RECEIPT_OR_INDICES
         | col::PROMISE_YIELD_INDICES
-        | col::PROMISE_YIELD_TIMEOUT => copy_kv_to_all_children(&status, key, value, store_update),
+        | col::PROMISE_YIELD_TIMEOUT
+        | col::BANDWIDTH_SCHEDULER_STATE => {
+            copy_kv_to_all_children(&status, key, value, store_update)
+        }
         col::BUFFERED_RECEIPT_INDICES | col::BUFFERED_RECEIPT => {
             copy_kv_to_left_child(&status, key, value, store_update)
         }
-        _ => unreachable!(),
+        _ => unreachable!("key: {:?} should not appear in flat store!", key),
```

Comment:
I wonder how we can make it so that when a new key is added the compiler would force the developer to support it here. If I recall correctly those are integers so it's not exactly possible to list all possible values in the match patterns. Perhaps you can find some clever way still to do it here using conditional patterns or something with the the max of col? If not then maybe a unit test would work? 

Not need to do it in this PR.

Reply:
Unit tests should catch it, as it happened in this case with bandwidth scheduler, once the new column starts to be used. Otherwise there can be some trick with `ALL_COLUMNS_WITH_NAMES`.. I'll think about that!